jmeter-plugins/src/main/java/plugins/jmeter/functions/XCorrelationIDHeader.java

```diff
@@ -0,0 +1,96 @@
+package plugins.jmeter.functions;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jmeter.engine.util.CompoundVariable;
+import org.apache.jmeter.functions.AbstractFunction;
+import org.apache.jmeter.functions.InvalidVariableException;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.samplers.Sampler;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.hashids.Hashids;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XCorrelationIDHeader extends AbstractFunction {
+    private Hashids hashids = new Hashids("XCorrelatiomnIDSalt", 8);
+    private static final String MDC_CORRELATION_ID_HEADER = "X-Correlation-ID";
+    private static final Logger log = LoggerFactory.getLogger(XCorrelationIDHeader.class);
+    private static final List<String> desc = new LinkedList<>();
+
+    private static final String KEY = "__XCorrelationIDHeader"; //$NON-NLS-1$
+
+    // Number of parameters expected - used to reject invalid calls
+    private static final int MIN_PARAMETER_COUNT = 1;
+    private static final int MAX_PARAMETER_COUNT = 2;
+
+    static {
+        desc.add(getResString("x_correlation_id_prefix")); //$NON-NLS-1$
+        desc.add(getResString("function_name_paropt")); //$NON-NLS-1$
+    }
+
+    private Object[] values;
+
+    public XCorrelationIDHeader() {
+
+    }
+
+    @Override
+    public List<String> getArgumentDesc() {
+        return desc;
+    }
+
+    @Override
+    public String execute(SampleResult previousResult, Sampler currentSampler) throws InvalidVariableException {
+        StringBuilder header = new StringBuilder();
+        header.append(MDC_CORRELATION_ID_HEADER);
+        header.append(": ");
+        String correlationId = generateCorrelationId();
+        String prefix = ((CompoundVariable) values[0]).execute().trim();
+        if (prefix.length() > 0) {
+            header.append(prefix);
+            header.append("-");
+        }
+        header.append(correlationId);
```

Comment:
it feels awkward that this code is only working with a delimited string. Instead, could you create a `org.apache.jmeter.protocol.http.control.Header` object, which keeps the header name/value in separate member variables? You could then use `vars.putObject` to store the `Header` as a variable, and finally you could return `header.toString()`. Otherwise, you are requiring any header managers to retrieve the variable as a String, parse the string (fragile!), and create their own `Header` objects.

Note that you'll have to include `ApacheJMeter_http` as a build dependency to have access to the `Header` class.

Alternately, just have this function be responsible for generating a unique id and optionally prefixing that id. It doesn't have to know anything about the `X-Correlation-ID` header, and in fact it doesn't care where the unique id is used. It just generates an optionally-prefixed unique id, and it is up to any callers to use that id wherever they please. This could make it useful outside of MDC, too.

Reply:
I like the alternative approach. We will just implement the random correlation id so users can just use this Jmeter function to get the id without worrying about the underlying implementation.